gateway/src/apicast/policy.lua

```diff
@@ -45,14 +47,18 @@ function _M.new(name, version)
     local mt = { __index = policy, __tostring = __tostring, policy = policy }
 
     function policy.new()
-        return setmetatable({}, mt)
+        local p = setmetatable({}, mt)
+
+        p.gc = GC.delegate(p)
```

Comment:
Why assign to `.gc` ?

Reply:
They have both pointer to the other, so their life is tied together.
When the `gc` object is GC'd it will call the `__gc` method on the policy metatable.

Using the `userdata` as proxy had significant performance issue in my early benchmarks, but can't reproduce it now, so they might have been invalid.

There are several other issues with the delegation on policies: calling `next` or `type`. Those are used by `inspect` and by some our code as well. We would have to patch `next` and `type` methods to rely on some other metamethods. 

I agree this is less than ideal and will explore the approach of allocating metatable per policy instance, so it can be GCd with the policy.